spec/features/register_for_competition_spec.rb

```diff
@@ -2,43 +2,68 @@
 
 require "rails_helper"
 
-RSpec.feature "Registering for a competition" do
+def reg_requirements_checkbox
+  all(:css, "label[for='regRequirementsCheckbox']").last
+end
+
+def find_modal(&)
+  all(:css, '.modal.visible', &).last
+end
+
+def within_modal(&)
+  within(find_modal(&))
+end
+
+RSpec.feature "Registering for a competition", js: true do
   let!(:user) { FactoryBot.create :user }
   let!(:delegate) { FactoryBot.create :delegate }
-  let(:competition) { FactoryBot.create :competition, :registration_open, :visible, delegates: [delegate] }
+  let(:competition) { FactoryBot.create :competition, :registration_open, :visible, :editable_registrations, delegates: [delegate] }
 
   context "signed in as user" do
     before :each do
       sign_in user
     end
 
-    scenario "User registers for a competition" do
+    scenario "User registers for a competition normally" do
       visit competition_register_path(competition)
-      check "registration_competition_events_333"
+      reg_requirements_checkbox.click
+      click_button "Continue to next Step"
+      click_button "checkbox-333"
       click_button "Register!"
-      expect(page).to have_text("Registration form submitted - make sure to follow all instructions from the Registration Requirements below to ensure your registration is complete.")
+      expect(page).to have_text("Your registration is processing...")
+      perform_enqueued_jobs
+      expect(page).to have_text("Your registration is pending approval by the organizers.")
       registration = competition.registrations.find_by_user_id(user.id)
-      expect(registration).not_to eq nil
+      expect(registration).not_to be_nil
     end
 
     scenario "User registers for a competition with invalid guest info" do
```

Comment:
EDIT: Especially given the test on Line 89, I suggest removing the guests part of this entirely and just making it a general "User submits invalid registration" smoketest

We have no way of knowing whether the registration is actually failing because of the invalid guest data - if events aren't selected, it may well just fail both times because the event selection is invalid. I think it's worth either asserting the presence of specific error text for guests, or just calling this "User can't register without selecting events

Reply:
In this test, we don't care why exactly the registration is nil. We just care _that_ it is nil, which we can be sure because we never selected any events.

I have renamed and commented more thoroughly, does that address your concerns?